packages/opds-web-client/src/utils/file.ts

```diff
@@ -0,0 +1,27 @@
+export const typeMap = {
+  "application/epub+zip": {
+    extension: ".epub",
+    name: "ePub"
+  },
+  "application/pdf": {
+    extension: ".pdf",
+    name: "PDF"
+  },
+  "application/vnd.adobe.adept+xml": {
+    extension: ".acsm",
+    name: "ACSM"
+  },
+  "application/x-mobipocket-ebook": {
+    extension: ".mobi",
+    name: "MOBI"
+  }
+};
+
+export const generateFilename = (str: string, extension: string) => {
```

Comment:
Can you write tests for this function now that it's a util function?
When you remove the `generateFilename` method from the DownloadButton component, a test should fail.

Reply:
Yep on it

Reply:
I didn't see any existing tests but I did write some of my own. Let me know if you're sufficient